### DISCUSSION TITLE
Saving pre-filled relation picker doesn't work

### DISCUSSION THREAD

**Discussion by arminus** [Budibase/budibase#11815] [Help] [answered]
So I have a form with a relation picker for a relation in an internal Budibase DB: customer relation from an invoice table to a customer table - which at this stage only has a name column).

My form saves the value of the chosen customer correctly when the customer is changed. But when I load the form with an existing relation (which is shown correctly in the form), the same save action sets the relation to empty in the table:

![SaveRelation](https://github.com/Budibase/budibase/assets/683680/50ebc0e2-3540-4792-9854-fd8f84f1c8ad)

So I looked at the post requests for the 2 different saves (with a modified and unmodified form), turns out the the content of _value.Customer and Customers are different:

For the modified form (where the save works):

![ModifiedForm-PostRequest](https://github.com/Budibase/budibase/assets/683680/20afa826-3c76-46a5-b888-e23b937237f2)

For the unmodified form (where the save doesn't work):

![UnModifiedForm-PostRequest](https://github.com/Budibase/budibase/assets/683680/1fe206ae-8903-4650-9aa1-151614b7cee7)

Bug?

**Comment by arminus** [comment]
This might be relate to https://github.com/Budibase/budibase/issues/10044 - but I'm using the internal DB.

**Comment by melohagan** [comment, marked as answer]
This bug has been fixed! https://github.com/Budibase/budibase/issues/11878
